### PR TITLE
[chore] 0.3.7 release

### PR DESCRIPTION
**Pull request by min-xu-ai**
## What does this PR do?
Fixes # (issue).

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by min-xu-ai** [comment]
There was a flaky test due to timing on the overlap test. Will need to do more fine tuning of the test to avoid flakiness.

**Review by blefaudeux — APPROVED** [review]
Nice, thank you @min-xu-ai !

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,20 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## NEXT - TBD
```
Comment (anj-s) [inline CHANGELOG.md:7]: Should we mention release number here?

**Review by anj-s — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (min-xu-ai) [reply CHANGELOG.md:7]: good catch. I should've updated it. fixing

**Pull request merged**